### PR TITLE
User update framework related test cases added

### PR DESCRIPTION
**Pull request by iostream04**

### CODE REVIEW

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 1]
```diff
@@ -49,15 +62,24 @@
...
 public class UserManagementActorTest {
 
   private ActorSystem system = ActorSystem.create("system");
   private static final Props props = Props.create(UserManagementActor.class);
 
   private static Map<String, Object> reqMap;
 
+  private static final String userId = "testUserId";
+  private static UserService userService;
+  private static SystemSettingClientImpl systemSettingClient;
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: this is no required to be declared at class level

**Comment by amolnin** [comment]
split the class with extends relationship - this is adding too many methods into one single class

**Comment by iostream04** [comment]
@amolnin  Please have a look now

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 2]
```diff
@@ -50,11 +51,12 @@
...
 })
-@PowerMockIgnore({"javax.management.*"})
+@PowerMockIgnore({"javax.management.*", "javax.crypto.*", "javax.net.ssl.*", "javax.security.*"})
 public class UserManagementActorTest {
 
-  private ActorSystem system = ActorSystem.create("system");
-  private static final Props props = Props.create(UserManagementActor.class);
+  public ActorSystem system = ActorSystem.create("system");
+  public static final Props props = Props.create(UserManagementActor.class);
+  public static UserService userService;
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: nothing should be declared public static unless it is final constant variable

File: service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DataCacheHandler.class, ContentStoreUtil.class, UserServiceImpl.class, Util.class})
+public class UserFrameworkUpdateTest extends UserManagementActorTest {
+
+  private static final String userId = "testUserId";
+  private static UserExternalIdentityDaoImpl userExtDao;
+
+  @Before
+  public void beforeEachTestn() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java]: beforeTest

File: service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  }
+
+  private void mockDatacacheHandler() {
+    Map<String, List<String>> frameworkFieldsConfigMap = new HashMap<>();
+    List<String> frameworkFieldConfig =
+        Arrays.asList("id", "medium", "gradeLevel", "board", "subject");
+    List<String> frameworkFieldConfigMan = Arrays.asList("id", "medium", "gradeLevel", "board");
+    frameworkFieldsConfigMap.put(JsonKey.FIELDS, frameworkFieldConfig);
+    frameworkFieldsConfigMap.put(JsonKey.MANDATORY_FIELDS, frameworkFieldConfigMan);
+    DataCacheHandler.setFrameworkFieldsConfig(frameworkFieldsConfigMap);
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java]: u don't need to use setters, just mock and return the wishful results from functions that return values.

File: service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    Mockito.when(DataCacheHandler.getFrameworkFieldsConfig()).thenReturn(frameworkFieldsConfigMap);
+
+    Map<String, List<Map<String, String>>> frameworkCategoriesMap = new HashMap<>();
+    frameworkCategoriesMap.put("medium", getListForCategoryMap("English"));
+    frameworkCategoriesMap.put("gradeLevel", getListForCategoryMap("Grade 3"));
+    frameworkCategoriesMap.put("board", getListForCategoryMap("NCERT"));
+
+    Map<String, Map<String, List<Map<String, String>>>> x = new HashMap<>();
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java]: proper names - what does x mean?

File: service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+import org.sunbird.learner.util.Util;
+import org.sunbird.user.dao.impl.UserExternalIdentityDaoImpl;
+import org.sunbird.user.service.impl.UserServiceImpl;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DataCacheHandler.class, ContentStoreUtil.class, UserServiceImpl.class, Util.class})
+public class UserFrameworkUpdateTest extends UserManagementActorTest {
+
+  private static final String userId = "testUserId";
+  private static UserExternalIdentityDaoImpl userExtDao;
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java]: Why mock our internal class -> u can mock cassandra operation, encryptionService for this purpose, no need to mock internal classes

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java (hunk 2 above)
Comment (amolnin) [reply service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: incorrect - we should not mock user-service it is internal class - using cassandra internally

**Comment by amolnin** [comment]
There seems to be problem in splitting the code:
Create one abstract class-> BaseUserActorTest (primarily keeps or defines methods/mocking common to both sub-classes)
Create one concrete class -> UserManagementActorTest (primarily adds any methods/mocking  - required for only User related test-cases)
Create another concrete class -> UserManagementActorFrameworkFieldsTest (primarly adds any method/mocking required for only Framework Field testing)


Let's discuss once you sketchout- how the class need to be divided. 

Also -> closing this request because it is showing changes other then test-classes. Please make sure you handle both before raising the new PR - No changes other than your test-case should reflect in this commit.

**Pull request closed without merging**